Backend/Controllers/UsersController.cs

```diff
@@ -0,0 +1,84 @@
+﻿using Backend.Models;
+using Backend.Models.Database;
+using Backend.Models.DTO;
+using JetBrains.Annotations;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Backend.Controllers
+{
+    [Route("api/v1/[controller]")]
+    [ApiController]
+    public class UsersController : ApiControllerBase
+    {
+        public UsersController(ApiContext context) : base(context) {}
+
+        [HttpGet]
+        public async Task<ActionResult<UserDTO>> GetUsers()
+        {
+            var users = await Context.Users
+                .Select(u => new GetUserDTO(u))
+                .ToListAsync();
+
+            return Ok(new GetUsersDTO(users));
+        }
+
+        [HttpGet("{id}")]
+        public async Task<ActionResult<GetPharmacyDTO>> GetUser(int id)
+        {
+            var user = await Context.Users
+                .FirstOrDefaultAsync(u => u.Id == id);
+
+            if (user == null) return ApiNotFound("User does not exist!");
+
+            return Ok(new GetUserDTO(user));
+        }
+
+        [HttpPost]
+        public async Task<ActionResult> AddUser([FromBody] CreateUserDTO dataFromBody)
+        {
+            ValidateCreateUserDTO(dataFromBody);
+
+            Context.Users.Add(new User(dataFromBody));
+            await Context.SaveChangesAsync();
+
+            return Created();
+        }
+
+        [HttpPut("{id}")]
+        public async Task<ActionResult> EditUser(int id, [FromBody] EditUserDTO dto)
+        {
+            var user = await Context.Users
+                .FirstOrDefaultAsync(u => u.Id == id);
+
+            if (user == null) return ApiNotFound("User does not exist!");
+
+            user.Name = dto.Name ?? user.Name;
```

Comment:
Maybe we could extract this logic to model?

Reply:
As in have a UserObj.UpdateProperties(EditUserDTO dto) method?

Reply:
Yes. Something like `UpdateFromDTO()`

Reply:
Moved.